src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.quarto.component.spec.ts

```diff
@@ -1049,6 +1119,143 @@ describe('OnlineGameWrapperComponent of Quarto:', () => {
+            expect(partDAO.update).toHaveBeenCalledOnceWith('joinerId', {
+                request: Request.localTimeAdded(Player.ONE),
+            });
+            const msUntilTimeout: number = (wrapper.joiner.maximalMoveDuration + 30) * 1000;
+            tick(msUntilTimeout);
+        }));
+        it('should add time to chrono local when receiving the addLocalTime request (Player.ONE)', fakeAsync(async() => {
+            // Given an onlineGameComponent
+            await prepareStartedGameFor(USER_CREATOR);
+            spyOn(partDAO, 'update').and.callThrough();
+            tick(1);
+
+            // when receiving addLocalTime request
+            await receivePartDAOUpdate({
+                request: Request.localTimeAdded(Player.ONE),
+            });
+
+            // then chrono local of player one should be filled
+            const wrapper: OnlineGameWrapperComponent = componentTestUtils.wrapper as OnlineGameWrapperComponent;
+            const msUntilTimeout: number = (wrapper.joiner.maximalMoveDuration + 30) * 1000;
+            expect(wrapper.chronoOneLocal.remainingMs).toBe(msUntilTimeout); // initial 2 minutes + 30 sec
+            tick(msUntilTimeout);
+        }));
+        it('should add time to local chrono when receiving the addLocalTime request (Player.ZERO)', fakeAsync(async() => {
+            // Given an onlineGameComponent
+            await prepareStartedGameFor(USER_CREATOR);
+            spyOn(partDAO, 'update').and.callThrough();
+            tick(1);
+
+            // when receiving addLocalTime request
+            await receivePartDAOUpdate({
+                request: Request.localTimeAdded(Player.ZERO),
+            });
+            // componentTestUtils.detectChanges(); // TODOTODO will we need this
+
+            // then chrono local of player one should be filled
+            const wrapper: OnlineGameWrapperComponent = componentTestUtils.wrapper as OnlineGameWrapperComponent;
+            const msUntilTimeout: number = (wrapper.joiner.maximalMoveDuration + 30) * 1000;
+            expect(wrapper.chronoZeroLocal.remainingMs).toBe(msUntilTimeout); // initial 2 minutes + 30 sec
+            tick(msUntilTimeout);
+        }));
+        it('should allow to add global time to opponent (as Player.ZERO)', fakeAsync(async() => {
+            // Given an onlineGameComponent
+            await prepareStartedGameFor(USER_CREATOR);
+            spyOn(partDAO, 'update').and.callThrough();
+            tick(1);
+
+            // when local countDownComponent emit addTime
+            await wrapper.addGlobalTime();
+
+            // then some kind of addGlobalTimeTo(player) should be sent
+            expect(partDAO.update).toHaveBeenCalledOnceWith('joinerId', {
+                request: Request.globalTimeAdded(Player.ONE),
+                remainingMsForOne: (1800 * 1000) + (5 * 60 * 1000),
+            });
+            const msUntilTimeout: number = wrapper.joiner.maximalMoveDuration * 1000;
```

Comment:
le GWT se répète plein de fois pour des tests différents, est-ce désiré ?

Reply:
J'ai un peu amélioré, mais c'est p-e toujours inoptimal !

Reply:
c'perturbant cet asymétrie dasn les noms: "addTurnTimeTo(player) should be sent" -> dans le code ça se traduit par "turnTimeAdded"

Reply:
Comment outdated ?

Reply:
wat?

Reply:
Est ce que le commentaires est hors-daté :D !